lib/piper/permissions/ast/chars/rule.ex

```diff
@@ -3,7 +3,12 @@ defimpl String.Chars, for: Piper.Permissions.Ast.Rule do
   alias Piper.Permissions.Ast
 
   def to_string(%Ast.Rule{command_selector: cs, permission_selector: ps}) do
-    "when #{cs} must have #{ps}"
+    "when #{cs} #{ps_str(ps)}"
   end
 
+  defp ps_str(%Ast.ConditionalExpr{op: :allow}=ps),
+    do: "#{ps}"
+  defp ps_str(ps),
+    do: "must have #{ps}"
```

Comment:
I wonder why the `to_string` function for `ConditionalExpr` doesn't already do this.